File: src/MedicalHospAPI/Models/Padrao.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedicalHospAPI.Models
{
    public class Padrao
    {
        [Key]
        public int Id { get; set; }

        [StringLength(255), Required]
        public string Nome { get; set; } = string.Empty;

        [StringLength(255), Required]
        public string Modelo { get; set; } = string.Empty;

        [StringLength(255), Required]
        public string Tag { get; set; } = string.Empty;

        [StringLength(255), Required]
        public string NumSerie { get; set; } = string.Empty;

        [Required]
        public bool Ativo { get; set; }

        [StringLength(255), Required]
        public string Patrimonio { get; set; } = string.Empty;

        [StringLength(255), Required]
        public string Setor { get; set; } = string.Empty;

        [Required, ForeignKey("Calibracao")]
        public int UltimaCalibracaoId { get; set; }
        public Calibracao UltimaCalibracao { get; set; }
    }
}

File: src/types/padrao.ts
import { Calibracao } from "./calibracao";

export type StatusCalibracao = "valido" | "atencao" | "vencido";

// export interface CalibracaoPadrao {
//     dataCalibracao: string;
//     proximoVencimento: string;
//     numeroCertificado: string;
//     frequencia: string;
//     fornecedor: string;
//     certificadoPdfUrl?: string;
//     status?: StatusCalibracao;
//     atualizadoEm?: string;
// }



export interface Padrao {
    id: string;
    nome: string;
    fabricante: string;
    modelo: string;
    tag: string;
    numSerie: string;
    patrimonio: string;
    setor: string;
    imagemUrl?: string;
    calibracao?: Calibracao;
}

export interface PadraoDTO {
    nome: string;
    fabricante: string;
    modelo: string;
    tag: string;
    numSerie: string;
    patrimonio: string;
    setor: string;
    imagemUrl?: string;
    calibracao?: Calibracao;
}